api/src/main/java/io/smallrye/reactive/messaging/annotations/Blocking.java

```diff
@@ -0,0 +1,47 @@
+package io.smallrye.reactive.messaging.annotations;
+
+import static java.lang.annotation.ElementType.METHOD;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Identifies that a particular method performs blocking operations,
+ * and as such should be executed on separate worker.
+ *
+ * When possible, the initial threads execution context is _restored_
+ * once the blocking operation has completed.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(METHOD)
+public @interface Blocking {
+    String DEFAULT_WORKER_POOL = "<no-value>";
+
+    /**
+     * Indicates the name of the worker pool to use for execution.
+     * By default all executions will be performed on the default worker pool.
+     *
+     * The maximum concurrency of a custom worker pool can be set with the following configuration key:
+     * <code>smallrye.messaging.worker.{pool-name}.max-concurrency</code>
+     *
+     * @return custom worker pool name for blocking execution.
+     */
+    String value() default DEFAULT_WORKER_POOL;
```

Comment:
For the value, is `.` allowed as part of the name?

Reply:
Probably not, just like other configs like topic name

Reply:
I was wrong, the way it's currently coded it can be any value

Reply:
Had a look at vertx.createSharedWorkerExecutor(). It seems there is no restriction on the name as far as javadoc is concerned.